src/inference/src/os/cpu_map_info.hpp

```diff
@@ -53,6 +53,39 @@ class CPU {
     std::map<int, int> _numaid_mapping_table;
     std::mutex _cpu_mutex;
     int _socket_idx = 0;
+
+private:
+    /**
+     * @brief      Sort proc_type_table by CPU ID on which application is running. The numa node containing this CPU ID
+     * will move to first row.
+     * @param[in]  _processor_id CPU ID on which application is running.
+     * @param[in] _proc_type_table summary table of number of processors per type
+     * @param[in] _cpu_mapping_table CPU mapping table for each processor
+     * @return
+     */
+    void sort_table_by_cpu_id(const int _processor_id,
+                              std::vector<std::vector<int>>& _proc_type_table,
+                              const std::vector<std::vector<int>>& _cpu_mapping_table) {
+        int current_numa_node = 0;
+        int current_socket = 0;
+
+        for (auto& row : _cpu_mapping_table) {
+            if (_processor_id == row[CPU_MAP_PROCESSOR_ID]) {
+                current_numa_node = row[CPU_MAP_NUMA_NODE_ID];
+                current_socket = row[CPU_MAP_SOCKET_ID];
+                break;
+            }
+        }
+        for (size_t i = 1; i < _proc_type_table.size(); i++) {
+            if ((current_numa_node == _proc_type_table[i][PROC_NUMA_NODE_ID]) &&
+                (current_socket == _proc_type_table[i][PROC_SOCKET_ID])) {
+                std::rotate(_proc_type_table.begin() + 1, _proc_type_table.begin() + i, _proc_type_table.end());
+                break;
+            }
+        }
+    };
+
+    friend class LinuxSortProcTableTests;
```

Comment:
Can test class as friend be removed?
Make `sort_table_by_cpu_id` as public or in test use test function to sort data.

Reply:
@praasz Dmitry ask to keep sort_table_by_cpu_id() as private function. 

@dmitry-gorokhov any comments on this?

Reply:
Yeah, I would prefer to avoid making functionality public just because we want to test it in unit tests. 
GoogleTests has some recommendations on it https://github.com/google/googletest/blob/main/docs/advanced.md#testing-private-code. Maybe FRIEND_TEST will work here.

Reply:
Yes, we just followed this GoogleTests recommendation. Since the private function in class CPU is defined in namespace ov, we follow the recommendation to use `friend class`.

Since the test function is only used inside the unit test file, "FRIEND_TEST" is not necessary.

@praasz @dmitry-gorokhov  If you do not have concern, this PR will be merged in following days.